pkg/adapterManager/manager.go

```diff
@@ -390,13 +390,15 @@ func newCacheKey(kind config.Kind, cfg *cpb.Combined) (*cacheKey, error) {
 
 	if cfg.Builder.Params != nil {
 		if err := enc.Encode(cfg.Builder.Params); err != nil {
+			pool.PutBuffer(b)
 			return nil, err
 		}
 		ret.builderParamsSHA = sha1.Sum(b.Bytes())
 	}
 	b.Reset()
```

Comment:
We don't need to remove in this PR, but this `Reset` isn't needed - `pool.PutBuffer` calls `Reset` on the buffer and is hit unconditionally after this is called.